src/main/java/com/allianceair/gims/service/ItemService.java

```diff
@@ -58,4 +86,24 @@ public Optional<InventoryItem> addServiceOrder(String id, ServiceOrder serviceOr
 
         return result;
     }
+
+    private List<InventoryDto> mapToDtos(Optional<List<InventoryItem>> inventory) {
+        return mapToDtos(inventory.get()); //TODO: read up on how to handle optional.ispresent == false
+    }
+
+    private List<InventoryDto> mapToDtos(List<InventoryItem> inventory) {
+        //NOTE: I dunno I do not love this, but I do not hate it. But this might be me mapping my dinosaur technology solutions to the new age
```

Comment:
Copying my note in here to make it more visible

I dunno I do not love this, but I do not hate it. But this might be me mapping my dinosaur technology solutions to the new age

Reply:
I'm actually unsure of this as well. I would need to do more research to understand the benefit of a DTO in a Document DB solution. 

Regardless, I think this is the place where Mapstruct would cut down on LOC.

Reply:
well I think if we switch to the names being unique and using them in the tables we can delay this decision. Thoughts?

Reply:
Category names being unique? Maybe worth a call sometime this week to get on the same page.

Reply:
yeah I agree with you. but Japan is kicking my ass again.

Any chance you got time on Saturday?

If not I can meet, but not during happy hour time. More like 8 in the evening.

Reply:
I just pushed a new alternative-normalization branch to each repo so you can see what I am thinking.

Reply:
I'll check out the next branch. I see why you chose this route after pulling the changes and running the code (the add item form was the give away).

I can try to make some time Saturday. I currently have nothing going on so pick a time and I'll make it work.